test_peek_zset_fetch_all and test_peek_zset_fetch_part failing with AssertionErrors

This is with iredis 1.12 on x86_64 and redis 7.0.2:
```
[   54s] =================================== FAILURES ===================================
[   54s] ___________________________ test_peek_zset_fetch_all ___________________________
[   54s] 
[   54s] iredis_client = <iredis.client.Client object at 0x7f06325923e0>
[   54s] clean_redis = Redis<ConnectionPool<Connection<host=localhost,port=6379,db=15>>>
[   54s] 
[   54s]     def test_peek_zset_fetch_all(iredis_client, clean_redis):
[   54s]         clean_redis.zadd(
[   54s]             "myzset", dict(zip([f"hello-{index}" for index in range(3)], range(3)))
[   54s]         )
[   54s]         peek_result = list(iredis_client.do_peek("myzset"))
[   54s] >       formatted_text_rematch(
[   54s]             peek_result[0][0:9],
[   54s]             FormattedText(
[   54s]                 [
[   54s]                     ("class:dockey", "key: "),
[   54s]                     ("", r"zset \(ziplist\)  mem: \d+ bytes, ttl: -1"),
[   54s]                     ("", "\n"),
[   54s]                     ("class:dockey", "zcount: "),
[   54s]                     ("", "3"),
[   54s]                     ("", "\n"),
[   54s]                     ("class:dockey", "members: "),
[   54s]                     ("", "\n"),
[   54s]                     ("", r"1\)"),
[   54s]                 ]
[   54s]             ),
[   54s]         )
[   54s] 
[   54s] tests/unittests/test_client.py:340: 
[   54s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   54s] 
[   54s] value_to_test = [('class:dockey', 'key: '), ('', 'zset (listpack)  mem: 96 bytes, ttl: -1'), ('', '\n'), ('class:dockey', 'zcount: '), ('', '3'), ('', '\n'), ...]
[   54s] expected_formatted_text = FormattedText([('class:dockey', 'key: '), ('', 'zset \\(ziplist\\)  mem: \\d+ bytes, ttl: -1'), ('', '\n'), ('class:dockey', 'zcount: '), ('', '3'), ('', '\n'), ('class:dockey', 'members: '), ('', '\n'), ('', '1\\)')])
[   54s] 
[   54s]     def formatted_text_rematch(value_to_test, expected_formatted_text):
[   54s]         """
[   54s]         ``expected_formatted_text`` can be regex.
[   54s]         """
[   54s]         for value, expected in zip(value_to_test, expected_formatted_text):
[   54s]             assert value[0] == expected[0]
[   54s]             print(expected[1], value[1])
[   54s] >           assert re.match(expected[1], value[1])
[   54s] E           AssertionError
[   54s] 
[   54s] tests/helpers.py:11: AssertionError
[   54s] ----------------------------- Captured stdout call -----------------------------
[   54s] key:  key: 
[   54s] zset \(ziplist\)  mem: \d+ bytes, ttl: -1 zset (listpack)  mem: 96 bytes, ttl: -1
[   54s] __________________________ test_peek_zset_fetch_part ___________________________
[   54s] 
[   54s] iredis_client = <iredis.client.Client object at 0x7f06324dee60>
[   54s] clean_redis = Redis<ConnectionPool<Connection<host=localhost,port=6379,db=15>>>
[   54s] 
[   54s]     def test_peek_zset_fetch_part(iredis_client, clean_redis):
[   54s]         clean_redis.zadd(
[   54s]             "myzset", dict(zip([f"hello-{index}" for index in range(40)], range(40)))
[   54s]         )
[   54s]         peek_result = list(iredis_client.do_peek("myzset"))
[   54s] >       formatted_text_rematch(
[   54s]             peek_result[0][0:8],
[   54s]             FormattedText(
[   54s]                 [
[   54s]                     ("class:dockey", "key: "),
[   54s]                     ("", r"zset \(ziplist\)  mem: \d+ bytes, ttl: -1"),
[   54s]                     ("", "\n"),
[   54s]                     ("class:dockey", "zcount: "),
[   54s]                     ("", "40"),
[   54s]                     ("", "\n"),
[   54s]                     ("class:dockey", r"members \(first 40\): "),
[   54s]                     ("", "\n"),
[   54s]                 ]
[   54s]             ),
[   54s]         )
[   54s] 
[   54s] tests/unittests/test_client.py:363: 
[   54s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   54s] 
[   54s] value_to_test = [('class:dockey', 'key: '), ('', 'zset (listpack)  mem: 560 bytes, ttl: -1'), ('', '\n'), ('class:dockey', 'zcount: '), ('', '40'), ('', '\n'), ...]
[   54s] expected_formatted_text = FormattedText([('class:dockey', 'key: '), ('', 'zset \\(ziplist\\)  mem: \\d+ bytes, ttl: -1'), ('', '\n'), ('class:dockey', 'zcount: '), ('', '40'), ('', '\n'), ('class:dockey', 'members \\(first 40\\): '), ('', '\n')])
[   54s] 
[   54s]     def formatted_text_rematch(value_to_test, expected_formatted_text):
[   54s]         """
[   54s]         ``expected_formatted_text`` can be regex.
[   54s]         """
[   54s]         for value, expected in zip(value_to_test, expected_formatted_text):
[   54s]             assert value[0] == expected[0]
[   54s]             print(expected[1], value[1])
[   54s] >           assert re.match(expected[1], value[1])
[   54s] E           AssertionError
[   54s] 
[   54s] tests/helpers.py:11: AssertionError
[   54s] ----------------------------- Captured stdout call -----------------------------
[   54s] key:  key: 
[   54s] zset \(ziplist\)  mem: \d+ bytes, ttl: -1 zset (listpack)  mem: 560 bytes, ttl: -1
[   54s] =============================== warnings summary ===============================
[   54s] iredis/commands.py:123
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/commands.py:123: DeprecationWarning: invalid escape sequence '\s'
[   54s]     re.search("\s", command)
[   54s] 
[   54s] iredis/commands.py:32
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/commands.py:32: DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     with open_text(project_data, "command_syntax.csv") as command_syntax:
[   54s] 
[   54s] iredis/commands.py:17
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/commands.py:17: DeprecationWarning: read_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     commands_summary = json.loads(read_text(project_data, "commands.json"))
[   54s] 
[   54s] ../../../../../usr/lib/python3.10/site-packages/importlib_resources/_legacy.py:80: 1 warning
[   54s] tests/unittests/test_client.py: 2 warnings
[   54s] tests/unittests/test_markdown_doc_render.py: 285 warnings
[   54s]   /usr/lib/python3.10/site-packages/importlib_resources/_legacy.py:80: DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     with open_text(package, resource, encoding, errors) as fp:
[   54s] 
[   54s] iredis/commands.py:52
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/commands.py:52: DeprecationWarning: open_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     with open_text(project_data, "dangerous_commands.csv") as dangerous_file:
[   54s] 
[   54s] iredis/redis_grammar.py:212
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/redis_grammar.py:212: DeprecationWarning: invalid escape sequence '\s'
[   54s]     COMMAND = "(\s*  (?P<command>[\w -]+))"
[   54s] 
[   54s] iredis/redis_grammar.py:224
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/redis_grammar.py:224: DeprecationWarning: invalid escape sequence '\d'
[   54s]     STREAM_ID = "(?P<stream_id>[T\d:>+*\-\$]+)"
[   54s] 
[   54s] iredis/redis_grammar.py:228
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/redis_grammar.py:228: DeprecationWarning: invalid escape sequence '\#'
[   54s]     SHARP_OFFSET = f"(?P<offset>\#?{NUM})"  # for bitfield command
[   54s] 
[   54s] tests/unittests/test_client.py::test_do_help
[   54s] tests/unittests/test_client.py::test_do_help
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/client.py:524: DeprecationWarning: read_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     doc = read_text(commands_data, f"{command_docs_name}.md")
[   54s] 
[   54s] tests/unittests/test_client.py: 1 warning
[   54s] tests/unittests/test_entry.py: 14 warnings
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/iredis/config.py:103: DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     with path(project_data, "iredisrc") as p:
[   54s] 
[   54s] tests/unittests/test_markdown_doc_render.py: 285 warnings
[   54s]   /home/abuild/rpmbuild/BUILD/iredis-1.11.1/tests/unittests/test_markdown_doc_render.py:27: DeprecationWarning: read_text is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
[   54s]     doc = read_text(commands_data, filename)
[   54s] 
[   54s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[   54s] =========================== short test summary info ============================
[   54s] FAILED tests/unittests/test_client.py::test_peek_zset_fetch_all - AssertionError
[   54s] FAILED tests/unittests/test_client.py::test_peek_zset_fetch_part - AssertionE...
[   54s] = 2 failed, 610 passed, 5 skipped, 4 deselected, 2 xfailed, 597 warnings in 40.01s =
```

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -13,7 +13,7 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         python: ['3.6', '3.7', '3.8', '3.9', '3.10']
-        redis: [5, 6]
+        redis: [5, 6, 7]
     runs-on: ${{ matrix.os }}
 
     services:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 ## UPCOMING
 
 - Feature: support new command: `HRANDFIELD`.
+- Bugfix: all tests pass on redis:7 now.
 
 ## 1.12
 

--- a/tests/unittests/test_client.py
+++ b/tests/unittests/test_client.py
@@ -20,9 +20,11 @@ from ..helpers import formatted_text_rematch
 def completer():
     return IRedisCompleter()
 
+
 zset_type = "ziplist"
-if os.environ['REDIS_VERSION'] == '7':
+if os.environ["REDIS_VERSION"] == "7":
     zset_type = "listpack"
+
 
 @pytest.mark.parametrize(
     "_input, command_name, expect_args",
@@ -173,13 +175,16 @@ def test_not_retry_on_authentication_error(iredis_client, config):
         iredis_client.execute("None", "GET", ["foo"])
 
 
-@pytest.mark.skipif("int(os.environ['REDIS_VERSION']) != 6", reason="""
+@pytest.mark.skipif(
+    "int(os.environ['REDIS_VERSION']) != 6",
+    reason="""
 in redis7, it will not work if you:
 1. connect redis without password
 2. set a password
 3. auth
 
-the auth will fail""")
+the auth will fail""",
+)
 def test_auto_select_db_and_auth_for_reconnect_only_6(iredis_client, config):
     config.retry_times = 2
     config.raw = True
@@ -347,7 +352,6 @@ def test_peek_zset_fetch_all(iredis_client, clean_redis):
         "myzset", dict(zip([f"hello-{index}" for index in range(3)], range(3)))
     )
     peek_result = list(iredis_client.do_peek("myzset"))
-
 
     formatted_text_rematch(
         peek_result[0][0:9],

--- a/tests/unittests/test_client.py
+++ b/tests/unittests/test_client.py
@@ -1,3 +1,4 @@
+import os
 import re
 import pytest
 import redis
@@ -19,6 +20,9 @@ from ..helpers import formatted_text_rematch
 def completer():
     return IRedisCompleter()
 
+zset_type = "ziplist"
+if os.environ['REDIS_VERSION'] == '7':
+    zset_type = "listpack"
 
 @pytest.mark.parametrize(
     "_input, command_name, expect_args",
@@ -343,12 +347,14 @@ def test_peek_zset_fetch_all(iredis_client, clean_redis):
         "myzset", dict(zip([f"hello-{index}" for index in range(3)], range(3)))
     )
     peek_result = list(iredis_client.do_peek("myzset"))
+
+
     formatted_text_rematch(
         peek_result[0][0:9],
         FormattedText(
             [
                 ("class:dockey", "key: "),
-                ("", r"zset \(ziplist\)  mem: \d+ bytes, ttl: -1"),
+                ("", rf"zset \({zset_type}\)  mem: \d+ bytes, ttl: -1"),
                 ("", "\n"),
                 ("class:dockey", "zcount: "),
                 ("", "3"),
@@ -371,7 +377,7 @@ def test_peek_zset_fetch_part(iredis_client, clean_redis):
         FormattedText(
             [
                 ("class:dockey", "key: "),
-                ("", r"zset \(ziplist\)  mem: \d+ bytes, ttl: -1"),
+                ("", rf"zset \({zset_type}\)  mem: \d+ bytes, ttl: -1"),
                 ("", "\n"),
                 ("class:dockey", "zcount: "),
                 ("", "40"),

--- a/tests/unittests/test_client.py
+++ b/tests/unittests/test_client.py
@@ -169,7 +169,13 @@ def test_not_retry_on_authentication_error(iredis_client, config):
         iredis_client.execute("None", "GET", ["foo"])
 
 
-@pytest.mark.skipif("int(os.environ['REDIS_VERSION']) < 6")
+@pytest.mark.skipif("int(os.environ['REDIS_VERSION']) != 6", reason="""
+in redis7, it will not work if you:
+1. connect redis without password
+2. set a password
+3. auth
+
+the auth will fail""")
 def test_auto_select_db_and_auth_for_reconnect_only_6(iredis_client, config):
     config.retry_times = 2
     config.raw = True

